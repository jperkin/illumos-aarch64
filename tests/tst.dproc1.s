/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2017 Joyent, Inc.
 */

/*
 * C3.5.7 Data-processing (1 source)
 */
.macro dproc op wreg=1
	/* Test 32-bit (except for rev32), 64-bit, and ZR */
.if \wreg
	\op	w0, w1
	\op	w30, wzr
.endif
	\op	x0, x1
	\op	x30, xzr
.endm
.text
.globl libdis_test
.type libdis_test, @function
libdis_test:
	/*
	 * C5.6.147 RBIT
	 */
	dproc rbit
	/*
	 * C5.6.150 REV16
	 */
	dproc rev16
	/*
	 * C5.6.149 REV
	 */
	dproc rev
	/*
	 * C5.6.151 REV32
	 */
	dproc rev32 0
	/*
	 * C5.6.40 CLZ
	 */
	dproc clz
	/*
	 * C5.6.39 CLS
	 */
	dproc cls
.size libdis_test, [.-libdis_test]
