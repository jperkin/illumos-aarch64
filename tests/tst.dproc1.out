    libdis_test:      20 00 c0 5a  rbit w0, w1
    libdis_test+0x4:  fe 03 c0 5a  rbit w30, wzr
    libdis_test+0x8:  20 00 c0 da  rbit x0, x1
    libdis_test+0xc:  fe 03 c0 da  rbit x30, xzr
    libdis_test+0x10: 20 04 c0 5a  rev16 w0, w1
    libdis_test+0x14: fe 07 c0 5a  rev16 w30, wzr
    libdis_test+0x18: 20 04 c0 da  rev16 x0, x1
    libdis_test+0x1c: fe 07 c0 da  rev16 x30, xzr
    libdis_test+0x20: 20 08 c0 5a  rev w0, w1
    libdis_test+0x24: fe 0b c0 5a  rev w30, wzr
    libdis_test+0x28: 20 0c c0 da  rev x0, x1
    libdis_test+0x2c: fe 0f c0 da  rev x30, xzr
    libdis_test+0x30: 20 08 c0 da  rev32 x0, x1
    libdis_test+0x34: fe 0b c0 da  rev32 x30, xzr
    libdis_test+0x38: 20 10 c0 5a  clz w0, w1
    libdis_test+0x3c: fe 13 c0 5a  clz w30, wzr
    libdis_test+0x40: 20 10 c0 da  clz x0, x1
    libdis_test+0x44: fe 13 c0 da  clz x30, xzr
    libdis_test+0x48: 20 14 c0 5a  cls w0, w1
    libdis_test+0x4c: fe 17 c0 5a  cls w30, wzr
    libdis_test+0x50: 20 14 c0 da  cls x0, x1
    libdis_test+0x54: fe 17 c0 da  cls x30, xzr
